### ISSUE TITLE
Drivetemp Renaming and One-HDD Bug

### ISSUE THREAD

**Issue by lasers** [su8/pinky-bar#4]
I'm separating issues because I can't tell which one you finished, but at same time, I'm not sure how I can separate bunch of suggestions (and feedback). Some of them would be appreciated and closed right away. Maybe the first issue to separate was a `loadavg` splitup.

Anyway, new issue.

1) Just like `--with-mpd`, I think it would be wise to use `--with-hddtemp` because I don't see why it was renamed to `--drivetemp` in first place. I assume staying with generic names was the reason.

2) I tried `smarttemp`. Not good because the letter is shared once again between two words which minimally give me (and would-be users) minor annoyance. I'm not sure what to do with `--with-smartemp`, but it seems like it could be its own argument if users prefer to use `smartmontools` instead of `hddtemp` instead of making users edit a script (untested, I will test this next).

2) Bug: `--drivetemp` prints only one hard drive. I have more than one. For `hddtemp`, strip off first + last characters in the line, then split line with `||` into hdds, then I guess, combine last two in every hdd. [py3status/hddtemp.py#L127-L132](https://github.com/ultrabug/py3status/blob/master/py3status/modules/hddtemp.py#L127-L132)

![drivetemp](https://user-images.githubusercontent.com/852504/38757344-4e8cf740-3f32-11e8-8a40-4783283e32f0.png)

EDIT: I got `smartemp` to work. This too give us one drive.

**Comment by su8** [comment]
Point 3 is resolved in https://github.com/su8/pinky-bar/commit/4f844993c5cacbace273a2d659e58c81bea5d729

**Comment by su8** [comment]
Just a small note that I have only 2 drives and the code in hddtemp may require the following line https://github.com/su8/pinky-bar/blob/4f844993c5cacbace273a2d659e58c81bea5d729/src/smart.c#L161 to use while loop, but don't have more drives to test out various scenarios.

```c
 /* hddtemp/daemon.c */
void daemon_send_msg(struct disk *ldisks, int cfd) {
struct disk * dsk;
daemon_update(ldisks, 0);
for(dsk = ldisks; dsk; dsk = dsk->next) {
  char msg[128];
```

**Comment by su8** [comment]
Point 1, well this one distinguishes hard drive and solid state drive, so we don't insult neither one of them. By calling it "drive" we specify either hard drive or ssd.

**Comment by lasers** [comment]
>By calling it "drive" we specify either hard drive or ssd.

https://en.wikipedia.org/wiki/Hard_disk_drive
>A hard disk drive (HDD), hard disk, hard drive or fixed disk[b] is a data storage device that uses magnetic storage to store...

Is it more accurate to call it `device` not to insult any new storage drives (or devices) in making? I don't have my hands on many toys so I don't know if `smartmontool` or `hddtemp` can read temperatures off on some devices other than drives.

>hddtemp

This works and not works. I `sort -u` this out of endless loops. `pinkybar` will print number of drives or devices, but mostly one single drive or device.
```
[\?color=yellow&show 53C ]
[\?color=yellow&show 53C 43C]
[\?color=yellow&show 53C 43C 52C]
[\?color=yellow&show 53C 43C 52C 34C ]
[\?color=yellow&show 53C 43C 52C 34C]
[\?color=yellow&show 53C 43C 52C 34C 41C]
```

**Comment by su8** [comment]
>This works and not works. I sort -u this out of endless loops. pinkybar will print number of drives or devices, but mostly one single drive or device.

Depends how often you call hddtemp, if you call it every second or less it will show this exactly, haven't looked much more in hddtemp daemon.c source code to see how often it gets updated, but can verify this is comming from hddtemp.

**Comment by lasers** [comment]
If I run `pinkybar` in quick endless loop, the result is different (in the log) than what I see. Realistically, I only see one... querying every 10 seconds. I would be very lucky to see it displaying a different number of drives.

**Comment by lasers** [comment]
Same result.

**Comment by su8** [comment]
I need more drives to solve this, don't have more than two.

**Comment by su8** [comment]
Hi @lasers can you test the latest commit and see if it's working fine **--with-drivetemp** ?

**Comment by su8** [comment]
Thank you for reporting this bug and further testing newer changes.

**Comment by su8** [comment]
Just a reminder to myself or anyone wanting to test many drives, create index.html and start python in `-mSimpleHTTPServer`, then change the port to 8000. Perform a simple query via `curl -s 127.0.0.1:8000`. The program should detect ~50 drives before it runs out of available characters to store the information.


The index.html

```bash
|/dev/sda|Corsair Force GT|26|C||/dev/sdc|Hitachi HDS721616PLA380|27|C||/dev/sda|Corsair Force GT|28|C||/dev/sdc|Hitachi HDS721616PLA380|29|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C| |/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C||/dev/sda|Corsair Force GT|30|C||/dev/sdc|Hitachi HDS721616PLA380|31|C|
```

**Comment by lasers** [comment]
What was the issue with 1 HDD bug? Out of available characters?

**Comment by su8** [comment]
The way curl transfers data is by splitting the data into smaller pieces/packages and calling the `read_curl_data_cb()` several times, previously it iterated the incoming data only once because the pieces/packages was small enough to fit when using 2 drives.
